src/coreComponents/linearAlgebra/DofManager.cpp

```diff
@@ -364,10 +487,28 @@ void DofManager::addCoupling( string const & fieldName,
 
   CouplingDescription & coupling = m_coupling[ {fieldIndex, fieldIndex} ];
   coupling.connector = Connector::Stencil;
-  coupling.regions = field.regions;
+  coupling.support = field.support;
   coupling.stencils = &stencils;
 }
 
+void DofManager::addCoupling( string const & rowFieldName,
+                              string const & colFieldName,
+                              DofManager::Connector connectivity,
+                              map< string, array1d< string > > const & regions,
+                              bool symmetric )
+{
+  // Convert input into internal format
+  std::vector< Regions > support;
+  for( auto const & p : regions )
+  {
+    MeshBody const & meshBody = m_domain->getMeshBody( p.first );
+    MeshLevel const & mesh = meshBody.getMeshLevel( 0 ); // TODO by name?
+    std::vector< string > regionNames( p.second.begin(), p.second.end() );
```

Comment:
I don't think I fully understand, could you explain a bit more? Note that I'm converting here from `array1d< string >` to `std::vector< string >` using the latter's range constructor. I don't think I can do more simple?

Reply:
Oh, sorry, I thought you could simply `std::vector< string > regionNames( p.second )` but I was confused.

Also, is there a reason why we are mixing `std::vector< string >` and `array1d< string >`?

Reply:
My reason is that `array1d` is missing some features of `vector`... mainly initializer-list construction which is very useful when writing [tests](https://github.com/GEOSX/GEOSX/blob/d34ec3645cbb0e88b1c4ff6436e041c80d45b0c0/src/coreComponents/unitTests/linearAlgebraTests/testDofManager.cpp#L659) and range constructor (granted, I could add those to `Array`... but I don't have time to). But also because `array1d<>` brings too many additional features and associated implementation complexity (e.g. dual memory space semantics) that aren't needed for a simple host-only linear container of strings. This isn't a super strong opinion... I could change internal storage to be `array1d` and add a compatibility API that accepts vector to keep tests simple. It's one way or the other.

On a related side note, if we keep using `LvArray::Array` to hold strings, we should at least use it with a `MallocBuffer` instead of `ChaiBuffer` to reduce unneeded overhead.